Improve documentation for list-commands (#3326)
ElektraInitiative/libelektra: doc/help/kdb-list-commands.md

## SYNOPSIS

> `kdb list-commands [OPTION]...`

Reviewer: This is inconsistent to the other man pages. We should either have `[OPTIONS]` everywhere or nowhere. The 3 dots at the end are misleading (there is not any other argument taken here).

Reviewer 2: How man pages work is clearly defined in [man(1)](http://man7.org/linux/man-pages/man1/man.1.html). It says, i.a.

> The command or function illustration is a pattern that should match all possible invocations.

What is currently in master does not do that, thus I added it.

>  [expression] ...   entire expression within [ ] is repeatable

And as far as I can tell using multiple options is possible: `kdb list-commands -C always -v -0`, thus using 3 dots looks correct to me.

My proposition adheres to the synopsis of GNU coreutils like [chown](http://man7.org/linux/man-pages/man1/chown.1.html), [chmod](http://man7.org/linux/man-pages/man1/chmod.1.html) and [ls](http://man7.org/linux/man-pages/man1/ls.1.html) and I think that we should apply this to other man pages, too.

Reviewer: It is unfortunately not consistently done across different tools, even not within the man tools themselves e.g. `whereis` no dots, `apropos` lists the possible options, ....

> thus using 3 dots looks correct to me.

rsync uses `[OPTION...]`

But this is actually not relevant, I was only complaining that we should consistent **within ourselves**. And we were consistent before, and this PR was the first to add `[OPTION]...`.

So either we add it everywhere, or leave it as is (which sounds to me the better way, as it is not really an important thing).

Reviewer 3: @Chemin1 could you please rebase and fix this little thing. It would be great to merge this work.

## DESCRIPTION

This command will list all internal kdb commands.
The output is suitable to be processed from other

> tools unless `--verbose` is specified.

Reviewer: ```suggestion
tools unless `-v` or `--verbose` is specified.
```

- `-C`, `--color <when>`:
  Print never/auto(default)/always colored output.
- `-v`, `--verbose`:

>   Print number of commands in the first line.

Reviewer: ```suggestion
  Prints the number of commands in the first line
```

>   Add a description to each command.

Reviewer: ```suggestion
  and adds a description to each command.
```

>   Print command names bold unless `--color never` is specified.

Reviewer: ```suggestion
  Prints command names in bold unless `--color never` is specified or output is printed to non-terminals.
```

- `-0`, `--null`:
  Use binary 0 termination

## EXAMPLES

> To print a list of all internal kdb commands with their description:

Reviewer: ```suggestion
To print a list of all internal kdb commands with their description use:
```
